### ISSUE TITLE
Eps is applied incorrectly to exp_avg_sq in apply_ams_bound(...)

### ISSUE THREAD

**Issue by AnatoliyGorkov** [kozistr/pytorch_optimizer#397]
Hello!

It appear to me that there is a bug in https://github.com/kozistr/pytorch_optimizer/blob/b0146c2ca652b47132404a08c51bbf5d3d218aa4/pytorch_optimizer/base/optimizer.py#L156-L175

Looks like there is in issue with how eps value is added to exp_avg_sq. For example, in a case where ams_bound is False, `de_nom = exp_avg_sq.add(eps).sqrt_().add_(eps)` and it should be `de_nom = exp_avg_sq.sqrt().add_(eps)` . Compared to default way it is done in Adam, this changes whey way that optimizers performs updates in case where gradient norm are small.

**Comment by kozistr** [comment]
> [@kozistr](https://github.com/kozistr) Got it, somehow I did not think that any negative values of `exp_avg_sq` were even possible.
> 
> In such case I suppose the following could be done:
> 
> 1. min-clipping `exp_avg_sq`, as you suggested, but I think some small `eps` is not needed there and it could just be min-clipped with zero.
> 2. The way it is currently done also seems like a solution, but my suggestion it to change the formula slightly. Instead of `de_nom = sqrt(exp_avg_sq + eps) + eps` it can be computed in a following way: `de_nom = sqrt(exp_avg_sq + eps ** 2) + eps` . Currently, in case of small `exp_avg_sq` values, the denominator in the update `~= sqrt(eps) + eps ~= sqrt(eps)`, since `eps` is some small value (typically 1e-8). And in a default case, typical gradient norms, at which there is some influence from adding eps is not ~= 1e-8, like in default Adam, but ~= 1e-4

thanks for your suggestion :)

> de_nom = sqrt(exp_avg_sq + eps ** 2) + eps

I agree with it's a decent solution. or maybe we can fix the eps value (e.g. 1e-15, 1e-16) to add to the `exp_avg_sq` in terms of reducing `**` computation.

I'll work on it sometime later (soon)! thanks for your idea

**Comment by AnatoliyGorkov** [comment]
Thank you!
